### PR TITLE
fixup! [op][agl][wayland] Properly adapt dimensions for web apps usin…

### PR DESCRIPTION
**Pull request by jdapena**
…g panel

On starting up applications, in case the application would not
override size, but previous application did, then it would take
previous application override. We should set override value to 0 if
previous application did not set it.

Bug-AGL: SPEC-3982
Signed-off-by: Jose Dapena Paz <jdapena@igalia.com>

### CODE REVIEW

File: src/agl/web_app_manager_service_agl.cc [hunk 1]
```diff
@@ -412,10 +412,8 @@ void WebAppManagerServiceAGL::LaunchStartupAppFromConfig() {
       [&](const auto& extension) { extensions_obj.append(extension); });
   obj["extensions"] = extensions_obj;
 
-  if (width)
-    width_ = atoi((const char*)width);
-  if (height)
-    height_ = atoi((const char*)height);
+  width_ = width ? atoi((const char*)width) : 0;
```
Comment (mv0) [inline src/agl/web_app_manager_service_agl.cc:415]: Wouldn't be clearer to just to init the values to 0?

**Comment by mv0** [comment]
Bit puzzled by the commit description, by "previous" does that mean this code is shared among other 
applications running? It kind seems like that, but yeah, a bit of surprise. Just to make sure I got this right.

**Comment by jdapena** [comment]
> Bit puzzled by the commit description, by "previous" does that mean this code is shared among other applications running? It kind seems like that, but yeah, a bit of surprise. Just to make sure I got this right.

Yes, this is part of the mess with how things communicate in the AGL service. An attempt to launch an application stores the data in certain variables, that are later used to actually launch the application.

Those variables were not cleaned before next launch attempt.

File: src/agl/web_app_manager_service_agl.cc (hunk 1 above)
Comment (jdapena) [reply src/agl/web_app_manager_service_agl.cc:415]: Values are initialized to 0 already. Problem is that service is shared among all the launch attempts, and there was no code resetting the values between attempts. Thus, we explicitely say in this code "if there is no value in the incoming launch flags, set to 0".

Comment (mv0) [reply src/agl/web_app_manager_service_agl.cc:415]: Yeah, I initial commented on this, and then saw the commit description, and added that comment. Fair enough.

**Comment by mv0** [comment]
LGTM.

**Pull request merged**